### PR TITLE
extend `csv-table` directive with filter logic

### PR DESCRIPTION
**Pull request by mikethebeer**
this plugin extends the `csv-table` directive from `docutils` package
with additional filter options.

### CODE REVIEW

File: src/crate/sphinx/csv.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                   # field delimiter char
+                   'delim': directives.single_char_or_whitespace_or_unicode,
+                   # treat whitespace after delimiter as significant
+                   'keepspace': directives.flag,
+                   # text field quote/unquote char:
+                   'quote': directives.single_char_or_unicode,
+                   # char used to escape delim & quote as-needed:
+                   'escape': directives.single_char_or_unicode,
+                   'filter_exclude': directives.unchanged,
+                   'filter_col_idx': directives.nonnegative_int,}
```
Comment (mikethebeer) [inline src/crate/sphinx/csv.py]: option_spec['filter_exclude'] = ''

File: src/crate/sphinx/csv.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    def run(self):
+        try:
```
Comment (mikethebeer) [inline src/crate/sphinx/csv.py]: remove

File: src/crate/sphinx/csv.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                csv_data, self.DocutilsDialect(self.options), source)
+
+            # filter row that does not contain a given string
+            if 'filter_exclude' in self.options:
+                filter_str = self.options['filter_exclude']
+                if 'filter_col_idx' in self.options:
+                    filter_col_idx = self.options['filter_col_idx']
+                    rows = self._apply_filter(rows, filter_str, filter_col_idx)
+                else:
+                    rows = self._apply_filter(rows, filter_str, None)
```
Comment (mikethebeer) [inline src/crate/sphinx/csv.py]: def parse_csv_data_into_rows(...)
    rows, max_cols = super()
    <code blcok>
    return rows, max_cols

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+If you have other extensions use:
+
+::
+
+    extensions.append('crate.sphinx.csv')
+
+First Example
+-------------
+
+::
```
Comment (bogensberger) [inline README.rst]: explain what it does, excludes third column which is "Attend".

File: docs/csv-table.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+=============
```
Comment (bogensberger) [inline docs/csv-table.txt]: move the content of this file to index.txt and delete this file

File: docs/index.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.. toctree::
+
+  csv-table
```
Comment (bogensberger) [inline docs/index.txt]: replace content with current content of index.txt

File: src/crate/sphinx/csv.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    def _apply_filter(self, rows, exclude_dict):
+        result = []
+
+        # append row that does not contain filter at column index
+        for row in rows:
+            exclude = False
+            for col_idx, pattern in exclude_dict.items():
+                # cell data value is located at hardcoded index pos. 3
+                # data type is always a string literal
+                val = row[col_idx][3][0]
```
Comment (bogensberger) [inline src/crate/sphinx/csv.py]: pls check that row has that index. Would throw a index out of bounds exception now, if idx > number of columns

**Comment by bogensberger** [comment]
LGTM

**Comment by aslanbekirov** [comment]
LGTM

**Pull request merged**